src/main/java/org/opensearch/knn/index/mapper/FlatVectorFieldMapper.java

```diff
@@ -73,13 +79,17 @@ private FlatVectorFieldMapper(
             stored,
             hasDocValues,
             indexCreatedVersion,
-            originalMappingParameters
+            originalMappingParameters,
+            isDerivedSourceEnabled
         );
         // setting it explicitly false here to ensure that when flatmapper is used Lucene based Vector field is not created.
         this.useLuceneBasedVectorField = false;
         this.perDimensionValidator = selectPerDimensionValidator(vectorDataType);
         this.fieldType = new FieldType(KNNVectorFieldMapper.Defaults.FIELD_TYPE);
         this.fieldType.setDocValuesType(DocValuesType.BINARY);
+        if (isDerivedSourceEnabled) {
```

Comment:
Are we adding only if it is true? is that intentional? if so why do we have DERIVED_VECTOR_FIELD_ATTRIBUTE_FALSE_VALUE ?

Reply:
Good point. Im going to remove FALSE in another PR.